File: use_case/interview/book-room.py
class BookRoom:
    def __init__(self, room_repository):
        self.room_repository = room_repository

    def book_room(self, room, start_date, end_date, interviewer, candidate):
        room.book(start_date, end_date, interviewer, candidate)
        self.room_repository.save(room)

File: src/core/use_case/book-room.ts
import {RoomRepositoryInterface} from "../business/room-repository.interface";
import {RoomDTO} from "../../common/dto/room.dto";
import {RoomMapper} from "../../common/mapper/room.mapper";

export class BookRoom {
  constructor(
    private readonly roomRepository: RoomRepositoryInterface,
    private readonly roomMapper: RoomMapper,
  ) {}

  book(room: RoomDTO, timeInterval: [Date, Date]): RoomDTO {
    const roomModel = this.roomMapper.toModel(room);

    roomModel.book(timeInterval);
    this.roomRepository.save(roomModel);

    return this.roomMapper.toDTO(roomModel);
  }
}
